src/controllers/UserController.ts

```diff
@@ -103,9 +98,7 @@ export class UserController {
     @Param('userID') userID: number,
     @CurrentUser({ required: true }) appUser: AppUser
   ): Promise<AppUserRolesResponse | undefined> {
-    const { role, id } = appUser;
-
-    if (!(role === AppUserRole.ADMIN || role === AppUserRole.OFFICER) && id != userID) {
+    if (this.appUserService.isInvalidNonOfficerAccess(appUser, userID)) {
```

Comment:
hey thanks for doing this, but probably can make the function name better - let's slap a fat TODO, open an issue and merge it

Reply:
Sounds good : )